keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -69,6 +78,36 @@ class ApplicationController(
     }
   }
 
+  @PostMapping(
+    path = ["/{application}/environment/{environment}/constraint"],
+    consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  @PreAuthorize("@authorizationSupport.userCanModifyApplication(#application)")
+  fun updateConstraintStatus(
+    @RequestHeader("X-SPINNAKER-USER") user: String,
+    @PathVariable("application") application: String,
+    @PathVariable("environment") environment: String,
+    @RequestBody status: UpdatedConstraintStatus
+  ) {
+    // we assume there's one manifest per application
+    val manifest = repository.getDeliveryConfigsByApplication(application).firstOrNull()
+      ?: throw NoDeliveryConfigForApplication(application)
+    val currentState = repository.getConstraintState(
+      manifest.name,
+      environment,
+      status.artifactVersion,
```

Comment:
a couple very tactical notes/questions about the current state of this endpoint:
1. How is this going to handle cases where there's multiple constraints of the same type? At the moment we have no way of uniquely identifying a constraint when there's multiple of the same type, and I expect we'll need that in the foreseeable future (not in this PR)
2. Is a version _on its own_ enough to globally identify a specific artifact version in a config? I realize at Netflix most debs/docker images have the artifact name encoded into the version, but in general my expectation/understanding was that you needed a tuple of name + type + version to uniquely identify a version within a config.

If one or both of those is worth a GitHub issue, I'll happily start one up. Again I doubt we'll need to worry about it in the next few weeks, but might in the next few months.

Reply:
Please start github issues for problems. This is strictly a port of the existing api. I'm happy to redesign functionality or improve it in the future